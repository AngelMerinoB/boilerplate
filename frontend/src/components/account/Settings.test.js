import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Settings } from './Settings';
Enzyme.configure({ adapter: new Adapter() });

describe('<Settings />', () => {
  test('should render <Settings /> without errors', () => {
    const wrapper = Enzyme.shallow(<Settings getAccount={() => ({})} />);
    expect(wrapper).toMatchSnapshot();
  });
});
