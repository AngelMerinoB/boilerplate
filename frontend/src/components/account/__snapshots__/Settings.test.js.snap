// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> should render <Settings /> without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Settings
    getAccount={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col-3"
        >
          <div
            className="list-group"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="list-group-item list-group-item-action"
              exact={true}
              to="/account/settings"
            >
              Profile
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="list-group-item list-group-item-action"
              to="/account/settings/change-password"
            >
              Change password
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="list-group-item list-group-item-action"
              to="/account/settings/2fa"
            >
              Security
            </NavLink>
          </div>
        </div>
        <div
          className="col-9"
        >
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/account/settings"
            />
            <Route
              component={[Function]}
              path="/account/settings/change-password"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/account/settings/2fa"
            />
            <Route
              component={[Function]}
              path="/account/settings/2fa/sms"
            />
            <Route
              component={[Function]}
              path="/account/settings/2fa/app"
            />
          </Switch>
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-3"
          >
            <div
              className="list-group"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                exact={true}
                to="/account/settings"
              >
                Profile
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                to="/account/settings/change-password"
              >
                Change password
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                to="/account/settings/2fa"
              >
                Security
              </NavLink>
            </div>
          </div>,
          <div
            className="col-9"
          >
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/account/settings"
              />
              <Route
                component={[Function]}
                path="/account/settings/change-password"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/account/settings/2fa"
              />
              <Route
                component={[Function]}
                path="/account/settings/2fa/sms"
              />
              <Route
                component={[Function]}
                path="/account/settings/2fa/app"
              />
            </Switch>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="list-group"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                exact={true}
                to="/account/settings"
              >
                Profile
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                to="/account/settings/change-password"
              >
                Change password
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                to="/account/settings/2fa"
              >
                Security
              </NavLink>
            </div>,
            "className": "col-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  exact={true}
                  to="/account/settings"
                >
                  Profile
                </NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  to="/account/settings/change-password"
                >
                  Change password
                </NavLink>,
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  to="/account/settings/2fa"
                >
                  Security
                </NavLink>,
              ],
              "className": "list-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Profile",
                  "className": "list-group-item list-group-item-action",
                  "exact": true,
                  "to": "/account/settings",
                },
                "ref": null,
                "rendered": "Profile",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Change password",
                  "className": "list-group-item list-group-item-action",
                  "to": "/account/settings/change-password",
                },
                "ref": null,
                "rendered": "Change password",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Security",
                  "className": "list-group-item list-group-item-action",
                  "to": "/account/settings/2fa",
                },
                "ref": null,
                "rendered": "Security",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/account/settings"
              />
              <Route
                component={[Function]}
                path="/account/settings/change-password"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/account/settings/2fa"
              />
              <Route
                component={[Function]}
                path="/account/settings/2fa/sms"
              />
              <Route
                component={[Function]}
                path="/account/settings/2fa/app"
              />
            </Switch>,
            "className": "col-9",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/account/settings"
                />,
                <Route
                  component={[Function]}
                  path="/account/settings/change-password"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/account/settings/2fa"
                />,
                <Route
                  component={[Function]}
                  path="/account/settings/2fa/sms"
                />,
                <Route
                  component={[Function]}
                  path="/account/settings/2fa/app"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/account/settings",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/account/settings/change-password",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/account/settings/2fa",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/account/settings/2fa/sms",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/account/settings/2fa/app",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-3"
          >
            <div
              className="list-group"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                exact={true}
                to="/account/settings"
              >
                Profile
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                to="/account/settings/change-password"
              >
                Change password
              </NavLink>
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="list-group-item list-group-item-action"
                to="/account/settings/2fa"
              >
                Security
              </NavLink>
            </div>
          </div>
          <div
            className="col-9"
          >
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/account/settings"
              />
              <Route
                component={[Function]}
                path="/account/settings/change-password"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/account/settings/2fa"
              />
              <Route
                component={[Function]}
                path="/account/settings/2fa/sms"
              />
              <Route
                component={[Function]}
                path="/account/settings/2fa/app"
              />
            </Switch>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-3"
            >
              <div
                className="list-group"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  exact={true}
                  to="/account/settings"
                >
                  Profile
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  to="/account/settings/change-password"
                >
                  Change password
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  to="/account/settings/2fa"
                >
                  Security
                </NavLink>
              </div>
            </div>,
            <div
              className="col-9"
            >
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/account/settings"
                />
                <Route
                  component={[Function]}
                  path="/account/settings/change-password"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/account/settings/2fa"
                />
                <Route
                  component={[Function]}
                  path="/account/settings/2fa/sms"
                />
                <Route
                  component={[Function]}
                  path="/account/settings/2fa/app"
                />
              </Switch>
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="list-group"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  exact={true}
                  to="/account/settings"
                >
                  Profile
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  to="/account/settings/change-password"
                >
                  Change password
                </NavLink>
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="list-group-item list-group-item-action"
                  to="/account/settings/2fa"
                >
                  Security
                </NavLink>
              </div>,
              "className": "col-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="list-group-item list-group-item-action"
                    exact={true}
                    to="/account/settings"
                  >
                    Profile
                  </NavLink>,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="list-group-item list-group-item-action"
                    to="/account/settings/change-password"
                  >
                    Change password
                  </NavLink>,
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="list-group-item list-group-item-action"
                    to="/account/settings/2fa"
                  >
                    Security
                  </NavLink>,
                ],
                "className": "list-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Profile",
                    "className": "list-group-item list-group-item-action",
                    "exact": true,
                    "to": "/account/settings",
                  },
                  "ref": null,
                  "rendered": "Profile",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Change password",
                    "className": "list-group-item list-group-item-action",
                    "to": "/account/settings/change-password",
                  },
                  "ref": null,
                  "rendered": "Change password",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Security",
                    "className": "list-group-item list-group-item-action",
                    "to": "/account/settings/2fa",
                  },
                  "ref": null,
                  "rendered": "Security",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/account/settings"
                />
                <Route
                  component={[Function]}
                  path="/account/settings/change-password"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/account/settings/2fa"
                />
                <Route
                  component={[Function]}
                  path="/account/settings/2fa/sms"
                />
                <Route
                  component={[Function]}
                  path="/account/settings/2fa/app"
                />
              </Switch>,
              "className": "col-9",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/account/settings"
                  />,
                  <Route
                    component={[Function]}
                    path="/account/settings/change-password"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/account/settings/2fa"
                  />,
                  <Route
                    component={[Function]}
                    path="/account/settings/2fa/sms"
                  />,
                  <Route
                    component={[Function]}
                    path="/account/settings/2fa/app"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/account/settings",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/account/settings/change-password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/account/settings/2fa",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/account/settings/2fa/sms",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/account/settings/2fa/app",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
